iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -528,7 +531,7 @@ bool PortManager::isCompatiblePubSub(const PublisherPortRouDiType& publisher,
     const bool historyRequestIsCompatible =
         !subOpts.requiresPublisherHistorySupport || subOpts.historyRequest <= pubOpts.historyCapacity;
 
-    return servicesMatch && blockingPoliciesAreCompatible && historyRequestIsCompatible;
+    return blockingPoliciesAreCompatible && historyRequestIsCompatible;
```

Comment:
We could also return early if the blocking policies are not compatible to be consistent. Would be more efficient. Whether we want the more expressive bool variables is a matter of taste, they will be optimized out in all likelihood.

Reply:
Already done on `master` in the meantime.